File: dotnet/01-commands/tests/LeapYearsKata/Year.cs

namespace LeapYearsKata
{
    public class Year
    {
        private int _year;
        public Year(int year)
        {
            _year = year;
        }
        public bool IsLeapYear()
        {
            if (IsYearDivisibleBy(400))
                return true;
            
            else if (IsYearDivisibleBy(4) && !IsYearDivisibleBy(100))
                return true;
            
            return false;
        }

        private bool IsYearDivisibleBy(int toDiviseBy)
        {
            return _year % toDiviseBy == 0;
        }
    }
}

File: python/01-commands/tests/LeapYearsKata/year.py
class Year:
    def __init__(self, year):
        self._year = year

    def is_leap_year(self):
        if self.is_year_divisible_by(400):
            return True
        elif self.is_year_divisible_by(4) and not self.is_year_divisible_by(100):
            return True
        else:
            return False

    def is_year_divisible_by(self, to_divide_by):
        return self._year % to_divide_by == 0